torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -6823,7 +6823,17 @@ std::tuple<Tensor, Tensor> index_reduce_backward(
         (grad * masked_src_result).index_select(dim, index),
         (grad * result).index_select(dim, index) /
             source.masked_fill(src_zero, 1));
-    if ((src_num_zeros > 1).any().item<bool>()) {
+    // is_meta(): meta tensors don't support .item(), so we can't do this check.
+    // The usual use case is for PT2, where we don't support double backward
+    // anyway - so if we can't verify src_num_zeros <= 0, then don't support
+    // double backward.
+    if (src_num_zeros.is_meta() || at::isTensorSubclassLike(src_num_zeros)) {
```

Comment:
@zou3519 the issue is that I added this line.

The goal of this line is to just assume that `src_num_zeros > 1` is true when using PT2, in order to avoid the .item() call. But `isTensorSubclassLike` also matches functorch tensors (and even if we just matched python subclass tensors that wouldn't be good enough, because there may be non-fake-tensor subclasses that want to track the actual behavior).

So I'm assuming this approach is wrong - any suggestions on a better way to handle this?

Reply:
Discussed with @albanD offline, it is easier to just branch on GradMode::is_enabled instead of `(src_num_zeros.is_meta() || at::isTensorSubclassLike(src_num_zeros)`. If we do not do double backwards, then GradMode::is_enabled is False, and we do not actually need to attach the DelayedError grad_fn.